jobbergate-api/jobbergate_api/apps/job_submissions/schemas.py

```diff
@@ -60,9 +60,218 @@
+    gpus_per_task: Optional[str] = Field(description="GPUs per task.")
+    hold: Optional[bool] = Field(
+        description="Specify the job is to be submitted in a held state (priority of zero)."
+    )
+    kill_on_invalid_dependency: Optional[bool] = Field(
+        description="If a job has an invalid dependency, then Slurm is to terminate it."
+    )
+    licenses: Optional[str] = Field(
+        description=(
+            "Specification of licenses (or other resources available on all nodes of the cluster) "
+            "which must be allocated to this job."
+        )
+    )
+    mail_type: Optional[str] = Field(description="Notify user by email when certain event types occur.")
+    mail_user: Optional[str] = Field(
+        description="User to receive email notification of state changes as defined by mail_type."
+    )
+    mcs_label: Optional[str] = Field(description="This parameter is a group among the groups of the user.")
+    memory_binding: Optional[str] = Field(description="Bind tasks to memory.")
+    memory_per_cpu: Optional[int] = Field(description="Minimum real memory per cpu (MB).")
+    memory_per_gpu: Optional[int] = Field(description="Minimum memory required per allocated GPU.")
+    memory_per_node: Optional[int] = Field(description="Minimum real memory per node (MB).")
+    minimum_cpus_per_node: Optional[int] = Field(description="Minimum number of CPUs per node.")
+    minimum_nodes: Optional[bool] = Field(
+        description="If a range of node counts is given, prefer the smaller count."
+    )
+    name: Optional[str] = Field(description="Specify a name for the job allocation.")
+    nice: Optional[str] = Field(description="Run the job with an adjusted scheduling priority within Slurm.")
+    no_kill: Optional[bool] = Field(
+        description="Do not automatically terminate a job if one of the nodes it has been allocated fails."
+    )
+    nodes: Optional[List[int]] = Field(
+        description="Request that a minimum of nodes nodes and a maximum node count.",
+        min_items=1,
+        max_items=2,
+    )
+    open_mode: Optional[Literal["append", "truncate"]] = Field(
+        description="Open the output and error files using append or truncate mode as specified."
+    )
+    partition: Optional[str] = Field(description="Request a specific partition for the resource allocation.")
+    priority: Optional[str] = Field(description="Request a specific job priority.")
+    qos: Optional[str] = Field(description="Request a quality of service for the job.")
+    requeue: Optional[bool] = Field(
+        description="Specifies that the batch job should eligible to being requeue."
+    )
+    reservation: Optional[str] = Field(
+        description="Allocate resources for the job from the named reservation."
+    )
+    signal: Optional[str] = Field(
+        description="When a job is within sig_time seconds of its end time, send it the signal sig_num.",
+        # regex=r"[B:]<sig_num>[@<sig_time>]",
```

Comment:
leftover?

Reply:
This is actually a placeholder for some further investigation. According to the documentation, this parameter has to follow a pattern:
```
"signal": {
    "pattern": "[B:]<sig_num>[@<sig_time>]",
    "type": "string",
    "description": "When a job is within sig_time seconds of its end time, send it the signal sig_num."
},
```

I'm wondering if we can check the pattern using some regex to inform Jobbergate's users about the error earlier in the workflow. As I said in the comment above, this was the first sub-task, and I need to work on a couple more to be able to perform integration tests and check it.

Reply:
solved in [235255a](https://github.com/omnivector-solutions/jobbergate/pull/189/commits/235255af07fb2a145bbcdcdf49abb795643a2b65)